packages/arb-avm-cpp/data_storage/include/data_storage/arbcore.hpp

```diff
@@ -178,12 +178,10 @@ class ArbCore {
    public:
     // Logs Cursor interaction
     bool logsCursorRequest(size_t cursor_index, uint256_t count);
-    std::optional<std::vector<value>> logsCursorGetLogs(
-        size_t cursor_index,
-        uint256_t& first_index_out);
-    std::optional<std::vector<value>> logsCursorGetDeletedLogs(
-        size_t cursor_index,
-        uint256_t& first_index_out);
+    std::pair<uint256_t, std::optional<std::vector<value>>> logsCursorGetLogs(
```

Comment:
I think a return type of `std::optional<std::pair<uint256_t, std::vector<value>>>` would be cleaner